crates/pint-submit/src/lib.rs

```diff
@@ -0,0 +1,59 @@
+use clap::builder::styling::Style;
+use essential_node_types::register_contract_solution;
+use essential_rest_client::builder_client::EssentialBuilderClient;
+use essential_types::{contract::Contract, solution::Solution, ContentAddress};
+use std::path::PathBuf;
+
+pub async fn submit_solution(
+    solution_opt: Option<Solution>,
+    builder_address: String,
+    contract_opt: Option<&Contract>,
+) -> anyhow::Result<ContentAddress> {
+    let solution: Solution = match (solution_opt, contract_opt) {
+        (Some(s), None) => s,
+        (None, Some(contract)) => {
+            let registry_predicate = essential_node_types::BigBang::default().contract_registry;
+            register_contract_solution(registry_predicate, contract)
+                ?
```

Comment:
currently getting the error:
```
`?` operator has incompatible types
`?` operator cannot convert from `SolutionData` to `Solution`
```
But this should be resolved after new crates are published.